File: datasets/lua-keep-incoder_6b_temp_0.2-2/HumanEval_56_correct_bracketing.lua
-- brackets is a string of "<" and ">".
-- return True if every opening bracket has a corresponding closing bracket.
-- >>> correct_bracketing("<")
-- False
-- >>> correct_bracketing("<>")
-- True
-- >>> correct_bracketing("<<><>>")
-- True
-- >>> correct_bracketing("><<>")
-- False
local function correct_bracketing(brackets)
  brackets = brackets:gsub("[<>]", "")
  local opening_brackets = brackets:gsub("^<", "")
  local closing_brackets = brackets:gsub("^>", "")
  local opening_brackets_count = opening_brackets:gsub("[^<>]+", ""):gsub("^<", "")
  local closing_brackets_count = closing_brackets:gsub("[^<>]+", ""):gsub("^>", "")
  return opening_brackets_count == closing_brackets_count
end


lu = require('luaunit')

function test_humaneval()
local candidate = correct_bracketing
    lu.assertEquals(candidate('<>'), true)
    lu.assertEquals(candidate('<<><>>'), true)
    lu.assertEquals(candidate('<><><<><>><>'), true)
    lu.assertEquals(candidate('<><><<<><><>><>><<><><<>>>'), true)
    lu.assertEquals(candidate('<<<><>>>>'), false)
    lu.assertEquals(candidate('><<>'), false)
    lu.assertEquals(candidate('<'), false)
    lu.assertEquals(candidate('<<<<'), false)
    lu.assertEquals(candidate('>'), false)
    lu.assertEquals(candidate('<<>'), false)
    lu.assertEquals(candidate('<><><<><>><>><<>'), false)
    lu.assertEquals(candidate('<><><<><>><>>><>'), false)
end

os.exit(lu.LuaUnit.run())

File: datasets/php-keep-code_davinci_001_temp_0.2-0/HumanEval_56_correct_bracketing.php
<?php
// brackets is a string of "<" and ">".
// return True if every opening bracket has a corresponding closing bracket.
// >>> correct_bracketing("<")
// False
// >>> correct_bracketing("<>")
// True
// >>> correct_bracketing("<<><>>")
// True
// >>> correct_bracketing("><<>")
// False
function correct_bracketing($brackets) {
    $stack = [];
    for ($i = 0; $i < strlen($brackets); $i++) {
        if ($brackets[$i] == "<") {
            array_push($stack, $brackets[$i]);
        } else {
            if (count($stack) == 0) {
                return False;
            }
            array_pop($stack);
        }
    }
    return count($stack) == 0;
}


function candidate(...$args) {
    return correct_bracketing(...$args);
}

function test(): void {
    if (candidate("<>") !== true) { throw new Exception("Test failed!"); }
    if (candidate("<<><>>") !== true) { throw new Exception("Test failed!"); }
    if (candidate("<><><<><>><>") !== true) { throw new Exception("Test failed!"); }
    if (candidate("<><><<<><><>><>><<><><<>>>") !== true) { throw new Exception("Test failed!"); }
    if (candidate("<<<><>>>>") !== false) { throw new Exception("Test failed!"); }
    if (candidate("><<>") !== false) { throw new Exception("Test failed!"); }
    if (candidate("<") !== false) { throw new Exception("Test failed!"); }
    if (candidate("<<<<") !== false) { throw new Exception("Test failed!"); }
    if (candidate(">") !== false) { throw new Exception("Test failed!"); }
    if (candidate("<<>") !== false) { throw new Exception("Test failed!"); }
    if (candidate("<><><<><>><>><<>") !== false) { throw new Exception("Test failed!"); }
    if (candidate("<><><<><>><>>><>") !== false) { throw new Exception("Test failed!"); }
}

test();